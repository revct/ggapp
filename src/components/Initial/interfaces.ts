import { NavigationParams } from "react-navigation";
import { NavigationStackScreenProps } from "react-navigation-stack";

export interface InitialProps extends NavigationStackScreenProps<NavigationParams> {
};

export interface InitialState {};
