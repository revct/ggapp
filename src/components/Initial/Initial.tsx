import React from 'react';
import {View, StyleSheet} from 'react-native';
import {InitialProps, InitialState} from './interfaces';
import { sleep } from 'utils/Misc';
import Spinner from 'components/Spinner/Spinner';

export class Initial extends React.Component<InitialProps, InitialState> {

  componentDidMount() {
    this.setAuthenticated();
  }

  async setAuthenticated() {
    const {navigation} = this.props;
    await sleep(1400);
    // navigate to guest stack
    return navigation.navigate('AppStack');
  }

  render(){
    return (
      <View style={styles.container}>
        <Spinner />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  }
});


export default Initial;
