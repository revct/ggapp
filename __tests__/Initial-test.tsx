import 'react-native';
import React from 'react';
import {act, create, ReactTestRenderer} from 'react-test-renderer';
import { Initial } from 'components/Initial/Initial';
import { NavigationParams } from 'react-navigation';

describe('Initial Screen', () => {
  describe('Rendering', () => {
    it('should render correctly', () => {
      let component:ReactTestRenderer;
      act(() => {
        component = create(<Initial
          auth={{
            token: null,
            setToken: jest.fn((token:string) => {})
          }}
          navigation={{
            navigate: (options:NavigationParams):any => true
          }}
        />)
      })
      expect(component.toJSON()).toMatchSnapshot();
    })
  })
});
