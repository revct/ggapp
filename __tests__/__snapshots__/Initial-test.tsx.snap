// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Screen Rendering should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#616161"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
